lib/tower.ex

```diff
@@ -455,64 +455,46 @@ defmodule Tower do
     rescue
       e ->
         Event.from_exception(e, __STACKTRACE__, metadata: %{test: true})
-        |> report_event(async: false)
+        |> report_event()
 
         :ok
     end
   end
 
-  defp report_event(%Event{} = event, options \\ []) do
-    async = Keyword.get(options, :async, true)
+  def async(fun) do
+    Tower.TaskSupervisor
+    |> Task.Supervisor.start_child(fun)
+  end
 
+  defp report_event(%Event{} = event) do
     reporters()
     |> Enum.each(fn reporter ->
-      report_event(reporter, event, async)
+      report_event(reporter, event)
     end)
   end
 
-  defp report_event(reporter, %Event{reason: %ReportEventError{reporter: reporter}}, _) do
+  defp report_event(reporter, %Event{reason: %ReportEventError{reporter: reporter}}) do
     # Ignore so we don't enter in a loop trying to report to the same buggy reporter
     :ignore
   end
 
-  defp report_event(reporter, event, false) do
+  defp report_event(reporter, event) do
     try do
       reporter.report_event(event)
     rescue
       exception ->
-        raise ReportEventError,
-          reporter: reporter,
-          original: {:error, exception, __STACKTRACE__}
+        async(fn ->
```

Comment:
I guess this `async` is needed to not stop the telemetry handler

Reply:
So that, in the case two or more reporters are listed, one buggy reporter doesn't crash the other reporters: https://github.com/mimiquate/tower/blob/b833fb5/test/tower_test.exs#L569-L620.

Reply:
Plus you get the exception report of the buggy reporter, reported to the 2nd reporter hehe :-)